### ISSUE TITLE
Unchecked annotation note causes dmypy to exit with code 1 on a successful run

### ISSUE THREAD

**Issue by vtclose** [python/mypy#14101]
**Bug Report**

When the code triggers `annotation-unchecked` warnings, both `mypy` and `dmypy` print a green "success" line. However there's an unexpected difference between exit codes. Mypy exits with `0` as expected in a successful check, but dmypy exits with a `1`. This prevents me from using dmypy in automated scenarios.

**To Reproduce**

```python
# example.py
def example():
    a: int = 1
    print(a + "2")
```

```bash
$ mypy example.py; echo $?
example.py:2: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Success: no issues found in 1 source file
0 # Good, expected

$ dmypy run example.py; echo $?
Daemon started
example.py:2: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Success: no issues found in 44 source files
1 # Bad, unexpected
```

**Expected Behavior**

Successful runs should return zero exit code in both mypy and dmypy.

**Environment**

```bash
$ python -V
Python 3.10.8
$ mypy -V
mypy 0.991 (compiled: yes)
```

A fresh venv with latest mypy installed, no configuration files at all.

**Comment by neob91-close** [comment]
I'm on top of this.

**Comment by charlietsai** [comment]
Is there any way to make warnings fail if we wanted that? My team finds the `annotation-unchecked` useful but our CI builds continue to pass so we often miss the warnings and end up committing typed code that doesn't get checked.

**Comment by ilevkivskyi** [comment]
> Is there any way to make warnings fail if we wanted that?

I don't think there is a way, but I see how such a flag for mypy may be useful. I would suggest you open a separate issue for this. In the meantime I guess you can write a bash wrapper script that fails if mypy shows some notes.

**Comment by agerlic** [comment]
We've noticed a similar issue with MyPy 1.1.1, could it be a regression?

**Comment by mb-915** [comment]
> We've noticed a similar issue with MyPy 1.1.1, could it be a regression?

I also noticed a similar issue with 1.14.1 [^other_versions], but it only appears on the second run, after the daemon has been started.
I will create a new issue if I find time to create a minimal working example.

[^other_versions]: and a lot of previous versions, I have this issue for at least 6 month
